apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -225,21 +225,25 @@ impl Plugin for Telemetry {
                                 Err(e)
                             }
                             Ok(router_response) => {
-                                let is_not_success =
+                                let mut has_errors =
                                     !router_response.response.status().is_success();
                                 Ok(router_response.map(move |response_stream| {
                                     let sender = sender.clone();
                                     let ctx = ctx.clone();
 
                                     response_stream
                                         .map(move |response| {
-                                            let response_has_errors = !response.errors.is_empty();
+                                            if !response.errors.is_empty() {
+                                                has_errors = true;
+                                            }
 
-                                            if !matches!(sender, Sender::Noop) {
+                                            if !response.has_next.unwrap_or(false)
+                                                && !matches!(sender, Sender::Noop)
+                                            {
                                                 Self::update_apollo_metrics(
                                                     &ctx,
                                                     sender.clone(),
-                                                    is_not_success || response_has_errors,
+                                                    has_errors,
```

Comment:
Just want to check, is it possible for has_errors to return true on some parts of the response and false on others? Is this a last one wins scenario?

Reply:
it's an "if one of the payloads reports an error, we will consider the response an error scenario"